php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -66,19 +66,30 @@ class Message
     /**
      * @ignore
      */
-    public function __construct($desc = NULL)
+    public function __construct($data = NULL)
     {
         // MapEntry message is shared by all types of map fields, whose
         // descriptors are different from each other. Thus, we cannot find a
         // specific descriptor from the descriptor pool.
-        if (get_class($this) === 'Google\Protobuf\Internal\MapEntry') {
```

Comment:
Where is MapEntry handled now?

Reply:
This is handled in `initWithDescriptor`, which is called when `$desc` is a descriptor. I can only find two places `MapEntry` is created, [here](https://github.com/google/protobuf/blob/master/php/src/Google/Protobuf/Internal/Message.php#L355) and [here](https://github.com/google/protobuf/blob/master/php/src/Google/Protobuf/Internal/Message.php#L1188). In both cases, a descriptor is provided in the constructor.